File: pset6/readability.py
# Imports libraries.
import cs50

# Gets Input.
text = cs50.get_string("Text: ")

# Intializes variables.
letterCount = 0

# Checks letter count.
for letter in text:
    if letter.isalnum() == True:
        letterCount += 1

# Checks word count.
wordCount = len(text.split())

# Checks sentence count.
sentenceCount = 0
for sentence in text:
    if sentence == '.' or sentence == '?' or sentence == '!':
        sentenceCount += 1

# Creates average variables.
aveLettersPer100Words = letterCount * (100 / wordCount)
aveSentencesPer100Words = sentenceCount * (100 / wordCount)

# Calculate grade level.
gradeLevel = (.0588 * aveLettersPer100Words) - (.296 * aveSentencesPer100Words) - 15.8

# Returns grade level.
if gradeLevel < 0:
    print("Before Grade 1")
elif gradeLevel > 16:
    print("Grade 16+")
else:
    print("Grade", round(gradeLevel))

File: pset2/readability.c
//Gets libraries and establishes main class.
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main()
{
    // Gets input.
    string input = get_string("Text: ");

    // Initializes variables.
    float words = 1;
    float letters = 0;
    float sentences = 0;
    int len = strlen(input);

    // Creates for loop to check every letter of input.
    for (int i = 0; i < len; i++)
    {
        // Counts letters.
        if (isalpha(input[i]))
        {
            letters++;

        }

        // Counts words.
        else if (isspace(input[i]))
        {
            words++;

        }

        // Counts Sentences
        else if (input[i] == '.' || input[i] == '?' || input[i] == '!')
        {
            sentences++;

        }
    }

    // Defines variables that represent rates.
    float l = letters / words * 100;
    float s = sentences / words * 100;

    // Defines result of formula as variable and rounds it/
    float index = .0588 * l - .296 * s - 15.8;
    int grade = round(index);

    // Returns grade level and accounts for out of bounds errors.
    if (grade >= 1 && index <= 16)
    {
        printf("Grade %i\n", grade);
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");

    }
    else
    {
        printf("Before Grade 1\n");

    }






}